File: src/main/java/com/msg/gauth/global/annotation/logger/InjectionLogger.java
package com.msg.gauth.global.annotation.logger;

import java.lang.reflect.Field;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.boot.context.event.ApplicationStartedEvent;
import org.springframework.context.ApplicationListener;
import org.springframework.context.ConfigurableApplicationContext;
import org.springframework.stereotype.Component;

@Component
public class InjectionLogger implements ApplicationListener<ApplicationStartedEvent> {

    @Override
    public void onApplicationEvent(ApplicationStartedEvent event) {
        ConfigurableApplicationContext ac = event.getApplicationContext();
        String[] beanDefinitionNames = ac.getBeanDefinitionNames();
        for (String name : beanDefinitionNames) {
            Object bean = ac.getBean(name);
            String classPath = bean.getClass().getName();
            Logger log = LoggerFactory.getLogger(classPath);
            try {
                if(!classPath.contains("gauth")){
                    continue;
                }
                Class<?> beanClass = Class.forName(classPath);
                Field[] fields = beanClass.getDeclaredFields();
                for (Field field : fields) {
                    log4k log4k = field.getDeclaredAnnotation(log4k.class);
                    if(log4k != null){
                        field.setAccessible(true);
                        field.set(bean, log);
                    }
                }
            } catch (ClassNotFoundException e) {
                throw new RuntimeException(e);
            } catch (IllegalAccessException e) {
                throw new RuntimeException(e);
            }
        }
    }
}

File: src/main/kotlin/com/msg/gauth/global/annotation/logger/InjectionLogger.kt
package com.msg.gauth.global.annotation.logger

import org.slf4j.LoggerFactory
import org.springframework.boot.context.event.ApplicationStartedEvent
import org.springframework.context.ApplicationListener
import org.springframework.stereotype.Component

@Component
class InjectionLogger: ApplicationListener<ApplicationStartedEvent> {
    override fun onApplicationEvent(event: ApplicationStartedEvent) {
        val ac = event.applicationContext
        val beanDefinitionNames = ac.beanDefinitionNames
        for (name in beanDefinitionNames) {
            val bean = ac.getBean(name!!)
            var classPath = bean.javaClass.name
            try {
                if (!classPath.contains("gauth"))
                    continue
                if(classPath.contains("$"))
                    classPath = classPath.substring(0,classPath.indexOf("$"))
                val beanClass = Class.forName(classPath)
                val fields = beanClass.declaredFields
                for (field in fields) {
                    val log4k = field.getDeclaredAnnotation(log4k::class.java)
                    if (log4k != null) {
                        val log = LoggerFactory.getLogger(classPath)
                        field.isAccessible = true
                        field[bean] = log
                    }
                }
            } catch (e: ClassNotFoundException) {
                throw RuntimeException(e)
            } catch (e: IllegalAccessException) {
                throw RuntimeException(e)
            }
        }
    }
}